lib/core/data/shared_preferences_repository.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+class SharedPreferencesRepository {
+  final SharedPreferences prefs;
+
+  SharedPreferencesRepository(this.prefs);
+
+  String? get playerName => prefs.getString('playerName');
+
+  bool? get enabledSound => prefs.getBool('enabledSound');
+
+  bool? get enabledMusic => prefs.getBool('enabledMusic');
+
+  set enabledMusic(bool? enabledMusic) => prefs.setBool('enabledMusic', enabledMusic ?? true);
+
+  set enabledSound(bool? enabledSound) => prefs.setBool('enabledSound', enabledSound ?? true);
```

Comment:
ambiguous name clash. You have to clarify how does the enabledMusic is different from the enabledSound. Sound is everything your phone emits, maybe try with enabledSoundEffects (or better soundEffectsEnabled)

Reply:
done